src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java

```diff
@@ -84,8 +84,8 @@ public static void processEnvironmentVariable(
             boolean rubyOpts) throws CommandLineException {
 
         String value = System.getenv(name);
-        if (value != null && value.length() != 0) {
-            List<String> args = Arrays.asList(value.split("\\s+"));
+        if (value != null && value.trim().length() != 0) {
+            List<String> args = Arrays.asList(value.trim().split("\\s+"));
```

Comment:
There was an issue with environment vars with whitespace: e.g. ` -Xgraal.warn_unless=false` would be split into:

`[" ","-Xgraal.warn_unless=false"]`

Reply:
Why is there a space in that environment variable?
In general, we should preserve spaces in ENV vars, but for TRUFFLERUBYOPT maybe it's OK to trim?

Reply:
cc @pitr-ch

Reply:
I think this is ok for all ENV vars. It takes care off a bad split rather than ignoring a space in an option inside tha env var.

Reply:
I mean changing the behavior of
```
$A=' b ' ruby -e 'p ENV["A"]'                              
" b "
```
would be unacceptable.
But since it's only done for RUBYOPT and TRUFFLERUBYOPT it seems fine.
(`RUBYOPT=' -rsocket ' ruby -e 'p Socket'` works on MRI)